pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/design/xml/ExceptionAsFlowControl.xml

```diff
@@ -95,4 +98,70 @@ class SubE extends TopE { }
 
         ]]></code>
     </test-code>
+    <test-code>
+        <description>#4434 simple propagation</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            import java.util.*;
+            public class Foo {
+
+                public Map<K, V> getAll(Iterable<? extends K> keys) throws ExecutionException {
+                    try {
+                        Map<K, V> result = cache.getAll(keys);
+                        if (something()) {
+                            throw new InvalidCacheLoadException("null key or value");
+                        }
+                        return result;
+                    } catch (NullPointerException | InvalidCacheLoadException e) {
+                        throw e; // <-- Violation
```

Comment:
I'm a little confused by this comment with `<expected-problems>0</expected-problems>`

Reply:
Oh yeah that comes from the original report
```suggestion
                        throw e;
```